### PR TITLE
feat: add integration tests for CLI commands

### PR DESCRIPTION
**Pull request by jesseturner21**
Add integration tests covering create (frameworks, memory, edge cases), add/remove resources, and validate commands with supporting test utilities.

## Description

Add integ tests for  
1. other frameworks create, checking project structure
2. edge cases for project name 
3. validate command
4. add and remove memory/identity

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update
- [x] Other (please describe): Tests

## Testing

How have you tested the change?

- [x] I ran `npm run test:all`
- [x] I ran `npm run typecheck`
- [x] I ran `npm run lint`
- [x] If I modified `src/assets/`, I ran `npm run test:update-snapshots` and committed the updated snapshots

## Checklist

- [ ] I have read the CONTRIBUTING document
- [ ] I have added any necessary tests that prove my fix is effective or my feature works
- [ ] I have updated the documentation accordingly
- [ ] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the
terms of your choice.

### CODE REVIEW

File: src/test-utils/prereqs.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    return false;
+  }
+}
+
+/**
+ * Check if AWS credentials are configured and valid.
+ */
+export function hasAwsCredentials(): boolean {
+  try {
+    execSync('aws sts get-caller-identity', { stdio: 'ignore' });
```
[the remark this answers is not in the document]
Comment (jesseturner21) [reply src/test-utils/prereqs.ts]: yes moved it out of prereq and it is just used in deploy

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -40,6 +40,7 @@ jobs:
       - run: npm ci
       - run: npm run build --if-present
       - run: npm run test:unit
+      - run: npm run test:integ
```
[the remark this answers is not in the document]
Comment (jesseturner21) [reply .github/workflows/build-and-test.yml]: added the skips in the test files.
